File: kotlin/modules/micronaut/src/main/kotlin/dev/zerosum/example/controller/GreetingController.kt
package dev.zerosum.example.controller

import dev.zerosum.example.greeting.GreetingService
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get

@Controller("/")
class GreetingController(
        private val greetingService: GreetingService
) {

    @Get
    fun hello(): String {
        return greetingService.greet()
    }
}

File: scala/modules/akka-http/src/main/scala/dev/zerosum/example/controller/GreetingController.scala
package dev.zerosum.example.controller

import dev.zerosum.example.greeting.GreetingService
import akka.http.scaladsl.model._
import akka.http.scaladsl.server.Directives._
import akka.http.scaladsl.server.Route

class GreetingController(val greetingService: GreetingService) {

  def route: Route = path("") {
    get {
      complete(
        HttpEntity(
          ContentTypes.`text/plain(UTF-8)`,
          greetingService.greet
        )
      )
    }
  }
}
